akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbStore.scala

```diff
@@ -6,25 +6,28 @@
 package akka.persistence.journal.leveldb
 
 import java.io.File
+
 import scala.collection.mutable
 import akka.actor._
 import akka.persistence._
-import akka.persistence.journal.{ WriteJournalBase }
+import akka.persistence.journal.WriteJournalBase
 import akka.serialization.SerializationExtension
 import org.iq80.leveldb._
 import scala.collection.immutable
 import scala.util._
 import scala.concurrent.Future
 import scala.util.control.NonFatal
 import akka.persistence.journal.Tagged
+import com.typesafe.config.Config
 
 /**
  * INTERNAL API.
  */
 private[persistence] trait LeveldbStore extends Actor with WriteJournalBase with LeveldbIdMapping with LeveldbRecovery {
-  val configPath: String
 
-  val config = context.system.settings.config.getConfig(configPath)
+  protected def transformConfig(cfg: Config) = cfg
+
+  val config = transformConfig(Persistence(context.system).configFor(self))
```

Comment:
I see that we use this in `AsyncWriteJournal` and it's not wrong, but I think the prescribed approach is to have it as a constructor parameter. See https://github.com/akka/akka-persistence-cassandra/blob/master/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala#L39